integration/test_timescaledb.py

```diff
@@ -294,15 +286,375 @@ async def test_timescaledb_async_emit(timescaledb):
+        controller.emit(test_data)
+        controller.terminate()
+        controller.await_termination()
+
+
+def test_timescaledb_schema_validation_with_schema_prefix(timescaledb):
+    """Test schema validation works correctly with schema.table format"""
+    (connection, _, _, _, _, _, table_name, dsn_url, _, _) = timescaledb
+
+    # Create a schema and table
     cursor = connection.cursor()
-    cursor.execute(f"SELECT COUNT(*) FROM {table_name}")
-    count = cursor.fetchone()[0]
+    schema_name = "test_schema"
+    cursor.execute(f"CREATE SCHEMA IF NOT EXISTS {schema_name};")
+
+    schema_table = f"{schema_name}.{table_name}_schema"
+    cursor.execute(f"DROP TABLE IF EXISTS {schema_table};")
+
+    cursor.execute(
+        f"""
+        CREATE TABLE {schema_table} (
+            time TIMESTAMPTZ NOT NULL,
+            required_col INTEGER NOT NULL,
+            optional_col VARCHAR(50)
+        );
+    """
+    )
+    cursor.execute(f"SELECT create_hypertable('{schema_table}', 'time');")
     cursor.close()
 
-    assert count == 2, "Async emission should insert exactly 2 rows"
+    time_format = "%d/%m/%y %H:%M:%S UTC%z"
+    controller = build_flow(
+        [
+            SyncEmitSource(),
+            TimescaleDBTarget(
+                dsn=dsn_url,
+                table=schema_table,  # Using schema.table format
+                time_col="time",
+                columns=["required_col", "optional_col"],
+                time_format=time_format,
+                max_events=10,
+            ),
+        ]
+    ).run()
+
+    # Test with missing required column
+    invalid_data = {
+        "time": "18/09/19 01:55:10 UTC+0000",
+        "optional_col": "test_value",
+        # Missing required_col
+    }
+
+    # Should raise ValueError for missing required column, showing correct schema.table format
+    with pytest.raises(ValueError, match=r"Missing required non-nullable column 'required_col'"):
+        controller.emit(invalid_data)
+        controller.terminate()
+        controller.await_termination()
+
+    # Cleanup
+    with connection.cursor() as cursor:
+        cursor.execute(f"DROP TABLE IF EXISTS {schema_table} CASCADE;")
+        cursor.execute(f"DROP SCHEMA IF EXISTS {schema_name} CASCADE;")
+
+
+@pytest.mark.asyncio
+async def test_timescaledb_schema_caching(timescaledb):
+    """Test that schema information is properly cached to avoid repeated queries"""
+    (_, _, _, _, _, _, table_name, dsn_url, timestamp_precision, columns_config) = timescaledb
+
+    time_formats = {"milliseconds": "%d/%m/%y %H:%M:%S.%f UTC%z", "microseconds": "%d/%m/%y %H:%M:%S.%f UTC%z"}
+    time_format = time_formats.get(timestamp_precision, "%d/%m/%y %H:%M:%S UTC%z")
+
+    target = TimescaleDBTarget(
+        dsn=dsn_url,
+        table=table_name,
+        time_col="time",
+        columns=columns_config,
+        time_format=time_format,
+        max_events=10,
+    )
+
+    # Initialize the target
+    await target._async_init()
```

Comment:
Here too, I think it would be better to test within a graph rather than by calling the target's private methods.

Reply:
will fix